examples/python/interceptors/headers/generic_client_interceptor.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2017 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Base class for interceptors that operate on all RPC types."""
+
+import grpc
+
+
+class GenericClientInterceptor(grpc.UnaryUnaryClientInterceptor,
+                               grpc.UnaryStreamClientInterceptor,
+                               grpc.StreamUnaryClientInterceptor,
+                               grpc.StreamStreamClientInterceptor):
+
+    def intercept_call(self, client_call_details, request_iterator, request_streaming, response_streaming):
+        return client_call_details, request_iterator, None
+
+    def intercept_unary_unary(self, continuation, client_call_details, request):
+        new_details, new_request_iterator, postprocess = self.intercept_call(client_call_details, iter((request,)), False, False)
+        response = continuation(new_details, next(new_request_iterator))
+        if postprocess:
+            return next(postprocess((response,)))
+        return response
+
+    def intercept_unary_stream(self, continuation, client_call_details, request):
+        new_details, new_request_iterator, postprocess = self.intercept_call(client_call_details, iter((request,)), False, True)
+        response_iterator = continuation(new_details, new_request_iterator)
+        if postprocess:
+            return postprocess(response_iterator)
+        return response_iterator
+
+    def intercept_stream_unary(self, continuation, client_call_details, request_iterator):
+        new_details, new_request_iterator, postprocess = self.intercept_call(client_call_details, request_iterator, True, False)
+        response = continuation(new_details, next(new_request_iterator))
+        if postprocess:
+            return next(postprocess(iter((response,))))
+        return response
+
+    def intercept_stream_stream(self, continuation, client_call_details, request_iterator):
+        new_details, new_request_iterator, postprocess = self.intercept_call(client_call_details, request_iterator, True, True)
+        response_iterator = continuation(new_details, new_request_iterator)
+        if postprocess:
+            return postprocess(response_iterator)
+        return response_iterator
```

Comment:
Tuck this into an `else:` (assumming "`return <conditional expression>`" doesn't fit on a single line).